Shared build of boost fails with undefined reference to __current_exception

Clean clone of mxe ([0886e91](https://github.com/mxe/mxe/commit/0886e918b77eabe411f73f60e0fdbad8990f8526))

```
$ git clone https://github.com/mxe/mxe /home/ikari/src/mxe2
$ make -C /home/ikari/src/mxe2 boost MXE_TARGETS=x86_64-w64-mingw32.shared
```
Ultimately results in a failed build due to a linker error: ``undefined reference to `__current_exception'``
The same happens when trying to build with `MXE_TARGETS=i686-w64-mingw32.shared`.
Static builds are working fine.

Possibly interesting excerpts from the build log:

(Probably unrelated, in configure stage)
```
    - zlib                     : yes [2]
    - bzip2                    : yes [2]
    - lzma                     : no [2]
    - zstd                     : no [2]
    - has_lzma_cputhreads builds : no [2]
error: No best alternative for /home/ikari/src/mxe2/tmp-boost-x86_64-w64-mingw32.shared/boost_1_89_0/libs/histogram/install-libraries-shared with <abi>ms <address-model>64 <architecture>x86 <asynch-exceptions>off <binary-format>pe <boost.beast.allow-deprecated>on <boost.beast.separate-compilation>on <boost.cobalt.executor>any_io_executor <boost.cobalt.pmr>std <context-impl>fcontext <coverage>off <debug-symbols>off <exception-handling>on <extern-c-nothrow>off <inlining>full <known-warnings>hide <link>shared <optimization>speed <os>LINUX <pch>on <preserve-test-targets>on <profiling>off <python-debugging>off <relevant>target-os <rtti>on <runtime-debugging>off <runtime-link>shared <stdlib>native <strip>off <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-gcc:version>mxe <toolset>gcc <variant>release <vectorize>off <visibility>hidden <warnings-as-errors>off <warnings>on <x-deduced-platform>x86_64
    no match: <local-visibility>hidden
    no match: <local-visibility>hidden <target-os>cygwin
    - cxx20_hdr_concepts       : no [2]
    - cxx20_hdr_concepts       : no [4]
```

The actual failure:
```
gcc.link.dll bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll
/home/ikari/src/mxe2/usr/bin/x86_64-w64-mingw32.shared-ld: bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/from_exception.o:from_exception:(.text+0x9b): undefined reference to `__current_exception'
/home/ikari/src/mxe2/usr/bin/x86_64-w64-mingw32.shared-ld: bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/from_exception.o:from_exception:(.text+0x27d): undefined reference to `__current_exception'
collect2: error: ld returned 1 exit status

    "x86_64-w64-mingw32.shared-g++"   -Wl,--out-implib,"bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll.a" -o "bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll"  -shared -Wl,--start-group "bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/from_exception.o"  -Wl,-Bstatic  -Wl,-Bdynamic  -Wl,--end-group -m64 -fvisibility=hidden -fvisibility-inlines-hidden 

...failed gcc.link.dll bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll.a...
...removing bin.v2/libs/stacktrace/build/gcc-mxe/release/x86_64/target-os-windows/threadapi-win32/threading-multi/visibility-hidden/libboost_stacktrace_from_exception-mt-x64.dll.a
```

I have attached the full log:
[boost_x86_64-w64-mingw32.shared.log](https://github.com/user-attachments/files/25121697/boost_x86_64-w64-mingw32.shared.log)

--- a/src/boost-1-fixes.patch
+++ b/src/boost-1-fixes.patch
@@ -5,7 +5,7 @@ Contains ad hoc patches for cross building.
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Tony Theodore <tonyt@logyst.com>
 Date: Wed, 8 Apr 2020 00:37:10 +1000
-Subject: [PATCH 1/1] fix darwin build
+Subject: [PATCH 1/2] fix darwin build
 
 
 diff --git a/tools/build/src/tools/darwin.jam b/tools/build/src/tools/darwin.jam
@@ -24,3 +24,34 @@ index 1111111..2222222 100644
      # - GCC 4.2 and higher in Darwin does not have -Wno-long-double.
      if [ version.version-less $(real-version) : 4 2 ]
      {
+
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
+Date: Wed, 11 Feb 2026 21:03:12 +0200
+Subject: [PATCH 2/2] Fix from_exception link on mingw
+
+Fixes the following linker error:
+from_exception:(.text+0x9b): undefined reference to `__current_exception'
+
+Source: https://github.com/boostorg/stacktrace/pull/225
+
+diff --git a/libs/stacktrace/build/Jamfile.v2 b/libs/stacktrace/build/Jamfile.v2
+index 1111111..2222222 100644
+--- a/libs/stacktrace/build/Jamfile.v2
++++ b/libs/stacktrace/build/Jamfile.v2
+@@ -29,6 +29,7 @@ lib dl ;
+ lib gcc_s ;
+ lib Dbgeng ;
+ lib ole32 ;
++lib ucrt ;
+ 
+ local LIBBACKTRACE_PATH = [ modules.peek : LIBBACKTRACE_PATH ] ;
+ lib backtrace
+@@ -264,6 +265,7 @@ lib boost_stacktrace_from_exception
+   : # requirements
+     <warnings>all
+     <target-os>linux:<library>dl
++    <target-os>windows:<library>ucrt
+ 
+     # Enable build when explicitly requested, or by default, when on x86
+     <conditional>@build-stacktrace-from-exception

